Content.Server/Lightning/LightningSystem.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System.Linq;
+using Content.Server.Beam;
+using Content.Server.Beam.Components;
+using Content.Server.Lightning.Components;
+using Content.Shared.Lightning;
+using Robust.Server.GameObjects;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Dynamics;
+using Robust.Shared.Random;
+
+namespace Content.Server.Lightning;
+
+public sealed class LightningSystem : SharedLightningSystem
+{
+    [Dependency] private readonly PhysicsSystem _physics = default!;
+    [Dependency] private readonly BeamSystem _beam = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LightningComponent, StartCollideEvent>(OnCollide);
+        SubscribeLocalEvent<LightningComponent, ComponentRemove>(OnRemove);
+    }
+
+    private void OnRemove(EntityUid uid, LightningComponent component, ComponentRemove args)
+    {
+        if (!TryComp<BeamComponent>(uid, out var lightningBeam)
+            || lightningBeam.VirtualBeamController == null
+            || !TryComp<BeamComponent>(lightningBeam.VirtualBeamController, out var beamController))
+        {
+            return;
+        }
+
+        beamController.CreatedBeams.Remove(uid);
+    }
+
+    private void OnCollide(EntityUid uid, LightningComponent component, StartCollideEvent args)
```

Comment:
```suggestion
    private void OnCollide(EntityUid uid, LightningComponent component, ref StartCollideEvent args)
```

After you merge to master.